lightpath/path.py

```diff
@@ -171,86 +176,81 @@ def __init__(self, *devices, name=None):
+        return [d for d in self.devices
+                if len(getattr(d, 'output_branches', ['1'])) > 1]
 
     @property
-    def range(self):
-        """
-        Starting z position of beamline
-        """
+    def range(self) -> Tuple[float, float]:
+        """ Tuple[float, float]: Starting z position of beamline """
         return self.path[0].md.z, self.path[-1].md.z
 
     @property
-    def path(self):
-        """
-        List of devices ordered by coordinates
-        """
+    def path(self) -> List[Device]:
+        """ List[Device]: List of devices ordered by coordinates """
         return sorted(self.devices, key=lambda dev: dev.md.z)
 
     @property
-    def blocking_devices(self):
+    def blocking_devices(self) -> List[Device]:
         """
         A list of devices that are currently inserted or are in unknown
         positions. This includes devices downstream of the first
         :attr:`.impediment`
+
+        Returns
+        -------
+        List[Device]
+            list of blockoing devices
         """
         # Cache important prior devices
-        prior = None
-        last_branches = list()
+        prev_device = None
+        prev_status = None
         block = list()
         for device in self.path:
-            # If we have switched beamlines
-            if prior and device.md.beamline != prior.md.beamline:
-                # Find improperly configured optics
-                for optic in last_branches:
-                    # If this optic is responsible for delivering beam
-                    # to this hutch and it is not configured to do so.
-                    # Mark it as blocking
-                    if device.md.beamline in optic.branches:
-                        if device.md.beamline not in optic.destination:
-                            block.append(optic)
-                    # Otherwise ensure it is removed from the beamline
-                    elif optic.md.beamline not in optic.destination:
-                        block.append(optic)
-                # Clear optics that have been evaluated
-                last_branches.clear()
-
-            # If our last device was an optic, make sure it wasn't required
-            # to continue along this beampath
-            elif (prior in last_branches
-                    and device.md.beamline in prior.branches
-                    and device.md.beamline not in prior.destination):
-                block.append(last_branches.pop(-1))
-
-            # Find branching devices and store
-            # They will be marked as blocking by downstream devices
-            dev_state = find_device_state(device)
-            if device in self.branches:
-                last_branches.append(device)
-            # Find inserted devices
-            elif dev_state == DeviceState.Inserted:
+            curr_state = find_device_state(device)
+            try:
+                curr_status = device.get_lightpath_status()
```

Comment:
It looks like `find_device_state` calls `get_lightpath_status`, so we're calling it twice to get the same information (however unlikely, it could also change between calls)

That helper probably should just return both the status and the state